src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -464,13 +465,13 @@ static inline void uct_dc_mlx5_iface_dci_alloc(uct_dc_mlx5_iface_t *iface, uct_d
     pool->stack_top++;
 }
 
-static inline void uct_dc_mlx5_iface_dci_free(uct_dc_mlx5_iface_t *iface,
-                                              uct_dc_mlx5_ep_t *ep)
+static UCS_F_ALWAYS_INLINE int
+uct_dc_mlx5_iface_dci_detach(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep)
```

Comment:
maybe pass "new_ep" param to dci_detach which could be either NULL or UCT_DC_IFACE_DCI_EP_DESTROYED?
this way can make sure that after this function, dci will no longer point to ep

Reply:
done